File: tuan02/bai38/bai38.py
def tang_nghiem_ngat(a: list[int]) -> bool:
    for i in range(1, len(a)):
        if a[i] <= a[i - 1]:
            return False
    return True

def khong_giam(a: list[int]) -> bool:
    for i in range(1, len(a)):
        if a[i] < a[i - 1]:
            return False
    return True

if __name__ == "__main__":
    import sys
    input_data = sys.stdin.read().split()
    if input_data:
        n = int(input_data[0])
        a = [int(x) for x in input_data[1:n+1]]
        print("YES" if khong_giam(a) else "NO")

File: tuan02/bai38/bai38.jl
function tang_nghiem_ngat(a::Vector{Int64})::Bool
    for i in 2:length(a)
        if a[i] <= a[i - 1]
            return false
        end
    end
    return true
end

function khong_giam(a::Vector{Int64})::Bool
    for i in 2:length(a)
        if a[i] < a[i - 1]
            return false
        end
    end
    return true
end

input_str = read(stdin, String)
tokens = split(input_str)
if !isempty(tokens)
    n = parse(Int64, tokens[1])
    a = [parse(Int64, tokens[i]) for i in 2:(n+1)]
    println(khong_giam(a) ? "YES" : "NO")
end